packages/devtools_app/lib/src/flutter/status_line.dart

```diff
@@ -2,26 +2,225 @@
+          ),
+        ),
+      );
+    } else {
+      // Use a placeholder for pages with no explicit documentation.
+      return Text('DevTools ${devtools.version}');
+    }
+  }
+
+  Widget buildPageStatus(
+      BuildContext context, Screen currentScreen, TextTheme textTheme) {
+    return currentScreen.buildStatus(context, textTheme);
+  }
+
+  Widget buildIsolateSelector(BuildContext context, TextTheme textTheme) {
+    final IsolateManager isolateManager = serviceManager.isolateManager;
+
+    // Listen to all isolate existence changes.
+    final Stream changeStream = combineStreams(
+      isolateManager.onSelectedIsolateChanged,
+      isolateManager.onIsolateCreated,
+      isolateManager.onIsolateExited,
+    );
+
+    return StreamBuilder<IsolateRef>(
+      initialData: isolateManager.selectedIsolate,
+      stream: changeStream.map((event) => isolateManager.selectedIsolate),
+      builder: (BuildContext context, AsyncSnapshot<IsolateRef> snapshot) {
+        final List<IsolateRef> isolates = isolateManager.isolates;
+
+        // When we have two or more isolates with the same name, append dome
+        // disambiguating information.
+        String disambiguatedName(IsolateRef ref) {
+          String name = ref.name;
+          if (isolates.where((e) => e.name == ref.name).length >= 2) {
+            name = '$name (${ref.number})';
+          }
+          return 'isolate: $name';
+        }
+
+        return DropdownButton<IsolateRef>(
+          value: snapshot.data,
+          onChanged: (IsolateRef ref) {
+            isolateManager.selectIsolate(ref?.id);
+          },
+          isDense: true,
+          items: isolates.map((IsolateRef ref) {
+            return DropdownMenuItem<IsolateRef>(
+              value: ref,
+              child: Text(disambiguatedName(ref), style: textTheme.bodyText2),
+            );
+          }).toList(),
+        );
+      },
+    );
+  }
+
+  Widget buildConnectionStatus(TextTheme textTheme) {
+    return StreamBuilder(
+      initialData: serviceManager.service != null,
+      stream: serviceManager.onStateChange,
+      builder: (BuildContext context, AsyncSnapshot<bool> snapshot) {
+        if (snapshot.data) {
+          final app = serviceManager.connectedApp;
+
+          String description;
+          if (!app.isRunningOnDartVM) {
+            description = 'web app';
+          } else {
+            final VM vm = serviceManager.vm;
+            description =
+                '${vm.targetCPU}-${vm.architectureBits} ${vm.operatingSystem}';
+          }
+
+          return Text(
+            'Connected ($description)',
+            style: textTheme.bodyText2,
+            overflow: TextOverflow.clip,
+            //maxLines: 1,
+          );
+        } else {
+          return Text(
+            'No client connection',
+            style: textTheme.bodyText2,
+          );
+        }
+      },
+    );
+  }
+}
+
+Stream combineStreams(Stream a, Stream b, Stream c) {
```

Comment:
nit: move this to a utils class. I'm surprised this isn't already in the collections package or similar.